Plots/WRF/NCL_WRF_zoom_1_2.py

```diff
@@ -0,0 +1,128 @@
+
+td2 = getvar(wrfin, "td2")
+
+###############################################################################
+# Create a subset of the data for zoomed in projection
+
+dims = td2.shape
+
+y_start = int(dims[0]/2)
+y_end = int(dims[0]-1)
+
+x_start = int(0)
+x_end = int(dims[1]/2)
+
+td2_zoom = td2[y_start:y_end, x_start:x_end]
+
+# Define the latitude and longitude coordinates
+lats, lons = latlon_coords(td2_zoom)
+
+###############################################################################
+# Plot the data
+
+# The `get_cartopy` wrf function will automatically find and use the 
+# intended map projection for this dataset 
+cart_proj = get_cartopy(td2_zoom)
+
+fig = plt.figure(figsize=(12,12))
+ax = plt.axes(projection=cart_proj)
+
+# Add features to the projection
+states = NaturalEarthFeature(category="cultural",
+                             scale="50m",
+                             facecolor="none",
+                             name="admin_1_states_provinces_shp")
+
+ax.add_feature(states, linewidth=0.5, edgecolor="black")
+ax.coastlines('50m', linewidth=0.8)
+
+# Add filled dew point temperature contours
+plt.contourf(to_np(lons),
+             to_np(lats),
+             to_np(td2_zoom),
+             levels=13, cmap="magma",
+             transform=ccrs.PlateCarree(),
+             vmin=-8,
+             vmax=18)
+
+# Add a colorbar
+cbar = plt.colorbar(ax=ax,
+                    orientation="horizontal",
+                    ticks=np.arange(-6,18,2),
+                    drawedges=True,
+                    extendrect=True,
+                    pad=0.08,
+                    shrink=0.75,
+                    aspect=30)
+
+# Format location of colorbar text to look like NCL version
+cbar.ax.text(0.5,
+             1.5,
+             '2m Dewpoint Temperature (C)',
+             horizontalalignment='center',
+             verticalalignment='center',
+             transform=cbar.ax.transAxes)
+
+# Format colorbar ticks and labels 
+cbar.ax.tick_params(labelsize=10)
+cbar.ax.get_xaxis().labelpad = -48
+
+# Draw gridlines
+gl = ax.gridlines(crs=ccrs.PlateCarree(),
+                  draw_labels=True,
+                  dms=False,
+                  x_inline=False,
+                  y_inline=False,
+                  linewidth=1,
+                  color="k",
+                  alpha=0.25)
+
+# Manipulate latitude and longitude gridline numbers and spacing
+gl.top_labels = False
+gl.right_labels = False
+gl.xlocator = mticker.FixedLocator([-120, -121, -122, -123, -124])
+gl.ylocator = mticker.FixedLocator([38, 39, 40, 41, 42])
+gl.xlabel_style = {"rotation": 0, "size": 15}
+gl.ylabel_style = {"rotation": 0, "size": 15}
+gl.xlines = False
+gl.ylines = False
+
+# Add titles to the plot
+plt.title("2m Dewpoint Temperature (C)", loc='left', y=1.02, size=12)
+plt.title("Zoomed in plot", loc='center', y=1.1, size=14)
+plt.title("OUTPUT FROM WRF V3.3.1 MODEL \nWE=481; SN=547; Levels=32; Dis=2km ; Phys Opt=2; PBL Opt=1; Cu Opt=0",
```

Comment:
Is any of the model information in the data file? I'm curious if we can parse out this info versus hard coding it in, that way someone using a different dataset won't have to change this line manually.

Reply:
So the answer is yes and no. The model information does exist, but I cannot find a way to format it in the way that we see in the NCL version. The problem comes (for me) from the combination of both text and `getattr` from the dataset. I've been playing around with different methods to extract the variables (xarray and netcdf Dataset) and tack them onto the end of text, but it seems like you can have one or the other, but not both. If you do have both, they will appear as a list such as: ['WE=', "481] or ('WE=', '481') without a way to be read as "WE=481" as is seen in the NCL version.

The version I am pushing now is the closest I could get which has the title of the projection taken from the data, but still hard codes the second line of data. Let me know what you think, or if you know of a way to combine text with data attributes.

Reply:
I'm not sure if I totally understand what issue you're having. Couldn't you pull the numbers from the file using `getattr` and then use the String `format()` method to mesh the data with the text you want? For example, say you have already pulled `WE` and `SN` from the file. You could do the following:
```
str_format = "WE={}; SN={};"
print(str_format.format(WE, SN))
```
Which prints out `WE=481; SN=547;`. You could expand this to include the other variables.